### PR TITLE
@W-22551291: Add stale-content-cleanup-inform admin workflow

### PR DESCRIPTION
**Pull request by Alon-ST-DATA**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

  _Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of                              
  the pull request._
                                                                                                                                    
  # Pull Request Template                                                                                                           
                       
  ## Description                                                                                                                    
                                                   
[plan doc](https://docs.google.com/document/d/1sHle1BHXn9Gmt_1U4MM-wW6VSMthyMaiTwDv2zbWZIE/edit?tab=t.0)
                                                                               
  Adds the JTBD #1 Inform-phase surface for the Tableau MCP admin-tools initiative: a way for Tableau Cloud site administrators to  
  identify stale workbooks and published datasources via MCP so they can drive a cleanup workflow.
                                                                                                                                    
  The change is gated end-to-end behind a new `TMCP_ADMIN_TOOLS_ENABLED` feature flag (default `false`) — when disabled, no new     
  tools or prompts are registered and the existing MCP surface is unchanged.
                                                                                                                                    
  **New MCP tools (admin-only):**                                                                                                   
                       
  - `get-stale-content-report` — deterministic, server-side anti-join of the Admin Insights `TS Events` (last-access events) and    
  `Site Content` (item universe) datasources. Applies the staleness threshold, optional project scope, and an Admin Insights project
   exclusion in TypeScript, then returns already-filtered rows. No client-side math.                                                
  - `query-admin-insights-ts-events` / `query-admin-insights-site-content` — thin admin-only wrappers over VDS for ad-hoc queries 
  against the two underlying Admin Insights datasources.                                                                            

  **New MCP prompt:**                                                                                                               
   
  - `stale-content-cleanup-inform` — first MCP prompt primitive in this repo. Thin shell that instructs the model to call           
  `get-stale-content-report` once with the threshold/project args and render the JSON response as a Markdown table for HITL review.
                                                                                                                                    
  **Cross-cutting infrastructure:**                                                                                               

  - `TMCP_ADMIN_TOOLS_ENABLED` feature flag (default `false`) gates all three tools and the prompt at registration time.
  - `adminGate` (`src/prompts/_lib/adminGate.ts`) verifies the caller's site role at request time via `GET
  /sites/{siteId}/users/{userId}` and rejects anything outside `SiteAdministratorCreator`, `SiteAdministratorExplorer`, or          
  `ServerAdministrator`. Result is cached for 5 minutes per `(siteId, userId)`.
  - Admin Insights LUID resolver with per-site cache.                                                                               
  - Admin tools intentionally bypass `resourceAccessChecker` — the Admin Insights datasources are internal/known and admin-gated  
  independently.                                                                                                                    
  - New overridable `STALE_CONTENT_MIN_AGE_DAYS` (default 90 — the Tableau Cloud TS Events lookback ceiling). Reuses the existing
  `INCLUDE_PROJECT_IDS` bound for project scoping; the tool also accepts an explicit `projectIds` argument (intersection applies    
  when both are set).                                                                                                             
  - New `AdminOnlyError` (403) and `AdminInsightsUnavailableError` (404) error classes.                                             
  - Bundle manifest entries for the two new env vars (`TMCP_ADMIN_TOOLS_ENABLED`, `STALE_CONTENT_MIN_AGE_DAYS`).                    
  - New `usersApi` (Zodios endpoint definition) + `usersMethods` (method class) for `GET /sites/{siteId}/users/{userId}`. The       
  endpoint is the only one needed by the admin gate today; `usersMethods` will be expanded in the follow-up license-reclamation     
  JTBD.                                                                                                                             
                                                                                                                                    
  **Design notes:**                                                                                                                 
   
  - The `Admin Insights` project itself is excluded from `get-stale-content-report` via a SET filter at the VDS query layer. Its    
  datasources are Tableau-managed and not user content, so surfacing them in cleanup reports is noise. The two raw query tools    
  remain unfiltered so admins can still query Admin Insights data directly when needed.                                             
  - Default threshold is **90 days** (not the monolith Stale Content view's 45) because TS Events on Tableau Cloud caps at a 90-day
  lookback — beyond that, items cannot be distinguished from each other on `daysSinceLastUse` without Advanced Management's 365-day 
  window.
  - Apply phase (tag → notify → delete), JTBD #2 (extract refresh optimization), and JTBD #3 (license reclamation) are out of scope 
  for this WI and will land in follow-ups.                                                                                          
   
  ## Motivation and Context                                                                                                         
                                                                                                                                  
  Tableau Cloud admins need a way to identify stale workbooks and published datasources via the Tableau MCP server so they can clean
   them up. The standard REST API only exposes `createdAt` / `updatedAt` — there is no `lastViewedAt` field on workbooks or
  datasources, and using `updatedAt` as a staleness proxy produces both false positives (workbooks viewed daily but never           
  re-published look "old") and false negatives (workbooks re-published yesterday but never viewed look "active").                 

  The fidelity-correct signal lives in **Admin Insights**: `TS Events` (Access events, 90-day Cloud window) anti-joined against     
  `Site Content` (universe of items including never-viewed). This change wraps that anti-join in a deterministic MCP tool and ships
  an MCP prompt that orchestrates it as an admin-driven workflow.                                                                   
                                                                                                                                  
  A first cut placed the anti-join + threshold filter inside the prompt itself (LLM-as-orchestrator). Manual mcpJam smoke testing   
  immediately surfaced a correctness issue: the model included a 75-day-stale item under a 90-day threshold (`75 > 90 → false`, but
  the item was rendered as stale anyway). The model self-corrected on challenge but produced a non-deterministic admin-facing       
  report. The shipping design moves all math into TypeScript inside `get-stale-content-report`; the prompt is a thin shell.       

  ## Type of Change

  - [ ] Bug fix
  - [x] New feature
  - [ ] Breaking change
  - [ ] Documentation update
  - [ ] Other (please describe):
                                                                                                                                    
  ## How Has This Been Tested?
                                                                                                                                    
  **Automated:**                                                                                                                  

  - `npx tsc --noEmit` — clean.                                                                                                     
  - `npm run lint` — clean.
  - `npm test -- --run` — **1188 / 1188 passing across 76 test files** (1148 pre-existing + 40 new). New tests:                     
    - **12** in `getStaleContentReport.test.ts` — anti-join math, threshold edge cases (incl. a regression test that locks the      
  75-day-vs-90-day mcpJam finding), never-accessed `COALESCE(last_access, Created At)` behavior, sort order, project scope filter,  
  admin-gate rejection, and three tests pinning the Admin Insights project exclusion at the VDS query layer (filter shape unit test,
   integration request-payload assertion, negative-control on `computeStaleRows`).                                                  
    - **6** in `staleContent/inform.test.ts` — prompt registration, feature-flag gating, tool-args round-trip for `minAgeDays` and
  `projectIds`, "do not recompute" instruction language.                                                                            
    - **5** in `_lib/adminGate.test.ts` — admin role passes, non-admin role rejected, missing role rejected, per-`(siteId, userId)`
  cache hits, cache invalidation.                                                                                                   
    - **3** in `resolver.test.ts` — name-to-LUID lookup, per-site cache, missing-dataset error.                                   
    - **3** in `queryTsEvents.test.ts` — tool name registration, VDS query routing through resolved LUID, admin-gate rejection.     
  - `npm run build` — clean (`build/index.js` ~1.8 MB).                                                                             
                                                                                                                                    
  **Manual (mcpJam against a Tableau Cloud Customer Zero site, `TMCP_ADMIN_TOOLS_ENABLED=true`, `STALE_CONTENT_MIN_AGE_DAYS=90`):** 
                                                                                                                                    
  - Initial smoke against the prompt-as-orchestrator design surfaced the 75-day false-positive issue. Driver for the                
  deterministic-tool refactor (Path B), now shipped.                                                                                
  - Re-test with the deterministic tool pending after this PR lands and the server is rebuilt — to be performed end-to-end before 
  enabling the flag in production.                                                                                                  
                                                                                                                                  
  ## Related Issues                                                                                                                 
                                                                                                                                  
  - W-22551291
                                                                                                                                    
  ## Checklist
                                                                                                                                    
  - [x] I have updated the version in the package.json file by using `npm run version`. For example,                              
        use `npm run version:patch` for a patch version bump. _(2.3.0 → 2.4.0 via `npm run version:minor`; this is a new feature, so
   a minor bump is appropriate.)_                                                                                                   
  - [x] I have made any necessary changes to the documentation _(tool description strings include the JSON output schema, caveats   
  about the 90-day TS Events lookback ceiling, and the Admin Insights project exclusion. Two new env vars are documented in the     
  Claude MCP bundle manifest.)_                                                                                                   
  - [x] I have added tests that prove my fix is effective or that my feature works _(40 new tests across 5 files; full suite still  
  green.)_                                                                                                                          
  - [x] New and existing unit tests pass locally with my changes _(1188 / 1188 passing.)_
  - [x] I have documented any breaking changes in the PR description. For example, renaming a config                                
        environment variable or changing its default value. _(No breaking changes. New surface is fully gated behind                
  `TMCP_ADMIN_TOOLS_ENABLED=false` by default; existing MCP surface and behavior are untouched when the flag is off.)_              
                                                                                                                                    
  ## Contributor Agreement                                                                                                          
                                                                                                                                  
  By submitting this pull request, I confirm that:

  - [x] I have read the [CONTRIBUTING guidelines](../../CONTRIBUTING.md) for this project and followed
        its Contribution Checklist.

### CODE REVIEW

**Review by Akash-Rastogi — APPROVED** [review]
LGTM

**Review by Akash-Rastogi — APPROVED** [review]

**Comment by Alon-ST-DATA** [comment]
Added [short demo ](https://drive.google.com/file/d/1vd1T44N9mTveT-E6HARKfW7NJRYDcFxC/view?usp=drive_link)using claude.  cc: @anyoung-tableau

File: docs/docs/extras/test-plan-admin-tools.md [hunk 1]
```diff
@@ -0,0 +1,557 @@
+---
+sidebar_position: 5
+title: Manual E2E Validation Plan — Admin Tools
+---
+
+# Tableau MCP for Admins — Manual E2E Validation Plan
```
Comment (anyoung-tableau) [inline docs/docs/extras/test-plan-admin-tools.md]: These docs are customer-facing so it feels a little out of place to include an internal test plan in them. Can you keep this in your team's Slack channel canvas or something?

File: docs/docs/tools/admin-insights/_category_.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "label": "Admin Insights",
+  "position": 7,
+  "link": {
+    "type": "generated-index",
+    "description": "Admin-only tools that query Tableau Cloud Admin Insights datasources via the VizQL Data Service. Gated by the `ADMIN_TOOLS_ENABLED` feature flag and a request-time site-role check."
```
Comment (anyoung-tableau) [inline docs/docs/tools/admin-insights/_category_.json]: We've generally referred to them as how they appear to users, so we should use the term `environment variable` instead of `feature flag`.

File: src/prompts/registry.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import { Config } from '../config.js';
+import { WebMcpServer } from '../server.web.js';
+
+export type WebPromptRegistration<Args extends ZodRawShape = ZodRawShape> = {
+  name: string;
+  title?: string;
+  description: string;
+  argsSchema?: Args;
+  // Returns true if the prompt should be registered for the current server config.
+  enabled: (config: Config) => boolean;
```
Comment (anyoung-tableau) [inline src/prompts/registry.ts]: Consider inverting this to `disabled` to be consistent with the `Tool` abstraction

File: src/tools/web/adminGate.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        `The caller's site role is "${observed}", which does not have permission to invoke admin-only tooling.`,
+    );
+    this.name = 'NotAdminError';
+  }
+}
+
+const cache: Map<string, { siteRole: string | undefined; expiresAt: number }> = new Map();
+const CACHE_TTL_MS = 5 * 60 * 1000;
+
+export const adminGate = {
```
Comment (anyoung-tableau) [inline src/tools/web/adminGate.ts]: I had asked Akosh to move this out of the _lib folder. Can we move it back to tools/web/adminGate.ts?

File: src/tools/web/adminInsights/getStaleContentReport.ts [hunk 5]
```diff
@@ -0,0 +1,429 @@
...
+const paramsSchema = {
+  minAgeDays: z
+    .number()
+    .int()
+    .min(1)
+    .max(3650)
+    .optional()
+    .describe(
+      'Minimum days since last access for content to be considered stale. ' +
+        'Defaults to the server-configured threshold (STALE_CONTENT_MIN_AGE_DAYS, default 90).',
```
Comment (anyoung-tableau) [inline src/tools/web/adminInsights/getStaleContentReport.ts]: It's not really helpful to the agent to inform it that the threshold can be configured on the server using the `STALE_CONTENT_MIN_AGE_DAYS` env var because the agent can't do anything about that. Any context provided in the argument descriptions or tool description should be as concise as possible while providing all the context the model needs in order to accurately invoke the tool.

File: docs/docs/extras/test-plan-admin-tools.md (hunk 1 above)
Comment (Alon-ST-DATA) [reply docs/docs/extras/test-plan-admin-tools.md]: Yes I was debating this. It makes total sense. Kept local copy and removed.

File: src/tools/web/adminInsights/getStaleContentReport.ts [hunk 6]
```diff
@@ -0,0 +1,429 @@
...
+              const siteContentResult = await executeAdminInsightsQuery({
+                restApi,
+                datasetName: ADMIN_INSIGHTS_DATASETS.SITE_CONTENT,
+                query: buildSiteContentQuery(types, projectNameScope),
+              });
+              if (siteContentResult.isErr()) {
+                return siteContentResult;
+              }
+
+              const universe = (siteContentResult.value.data ?? []) as SiteContentRow[];
```
Comment (anyoung-tableau) [inline src/tools/web/adminInsights/getStaleContentReport.ts]: We've made it a point to avoid type assertions if at all possible. Can we define a Zod schema for the siteContentRow instead and infer the `SiteContentRow` from it?

File: docs/docs/tools/admin-insights/_category_.json (hunk 2 above)
Comment (Alon-ST-DATA) [reply docs/docs/tools/admin-insights/_category_.json]: Revised.

File: src/tools/web/adminInsights/resolver.ts [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  TS_EVENTS: 'TS Events',
+  SITE_CONTENT: 'Site Content',
+} as const;
+
+export type AdminInsightsDataset =
+  (typeof ADMIN_INSIGHTS_DATASETS)[keyof typeof ADMIN_INSIGHTS_DATASETS];
+
+type SiteCacheEntry = Map<string, string>;
+
+const cache: Map<string, SiteCacheEntry> = new Map();
```
Comment (anyoung-tableau) [inline src/tools/web/adminInsights/resolver.ts]: This caching should look more like the caching in the `adminGate`. Currently it will grow unbounded until the box runs out of RAM.

Comment (Alon-ST-DATA) [reply src/tools/web/adminInsights/resolver.ts]: Correct and for that we have an optimization story:  [W-22551424](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00002aJhI3YAK/view) 
but let me look further if anything can be done now.

File: src/tools/web/adminGate.ts (hunk 4 above)
Comment (Alon-ST-DATA) [reply src/tools/web/adminGate.ts]: Done.

File: src/tools/web/adminInsights/getStaleContentReport.ts (hunk 5 above)
Comment (Alon-ST-DATA) [reply src/tools/web/adminInsights/getStaleContentReport.ts]: Agree. Removed.

File: src/tools/web/adminInsights/resolver.ts (hunk 7 above)
Comment (Alon-ST-DATA) [reply src/tools/web/adminInsights/resolver.ts]: Tracked the full cache redesign in W-22551424. 
For now I will mirror adminGate's ExpiringMap pattern — TTL-bounded entries, parity with the existing convention. The deeper questions (size limits, eviction policy, telemetry) move to the  optimization story.

File: src/tools/web/adminInsights/getStaleContentReport.ts [hunk 8]
```diff
@@ -0,0 +1,429 @@
...
+    const name = byId.get(id);
+    if (name) {
+      out.add(name);
+    }
+  }
+  return Array.from(out);
+}
+
+const PROJECT_NAME_CACHE_TTL_MS = 5 * 60 * 1000;
+const projectNameCache: Map<string, { byId: Map<string, string>; expiresAt: number }> = new Map();
```
Comment (anyoung-tableau) [inline src/tools/web/adminInsights/getStaleContentReport.ts]: This cache is worrisome since entries never get removed and could grow without bound. Can we use the `ExpiringMap` pattern similar to `adminGate`?

File: src/overridableConfig.ts [hunk 9]
```diff
@@ -62,6 +67,8 @@ export class OverridableConfig {
   disableQueryDatasourceValidationRequests: boolean;
   disableMetadataApiRequests: boolean;
 
+  staleContentMinAgeDays: number;
```
Comment (anyoung-tableau) [inline src/overridableConfig.ts:70]: Please update these docs for the new overridable value:

docs/docs/configuration/mcp-config/site-settings.md
docs/docs/configuration/mcp-config/request-overrides.md

File: src/tools/web/adminInsights/getStaleContentReport.ts (hunk 8 above)
Comment (Alon-ST-DATA) [reply src/tools/web/adminInsights/getStaleContentReport.ts]: agree. refined.

File: src/overridableConfig.ts (hunk 9 above)
Comment (Alon-ST-DATA) [reply src/overridableConfig.ts:70]: revised

**Review by anyoung-tableau — APPROVED** [review]
